### PR TITLE
Global folder integration for all features that need a baserom

### LINKED ISSUE #676: flips does not account for shared baseroms (#644, #670)
PR to address this coming soon, this is for tracking

### PR DESCRIPTION
**Pull request by someone2639**
includes flips and extracting from jp/eu/sh versions

(depends on #670 )
(closes #676 )

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -309,19 +309,19 @@ ifeq ($(filter clean distclean print-%,$(MAKECMDGOALS)),)
...
       endif
     endif
-    ifneq (,$(wildcard baserom.eu.z64))
+    ifneq (,$(shell python3 tools/detect_baseroms.py eu))
       DUMMY != $(PYTHON) extract_assets.py eu >&2 || echo FAIL
       ifeq ($(DUMMY),FAIL)
         $(error Failed to extract assets from EU ROM)
       endif
     endif
-    ifneq (,$(wildcard baserom.sh.z64))
```
Comment (gheskett) [inline Makefile:324]: Doesn't probably matter that much, but why aren't we forcing the top version to be US always here?

**Review by gheskett — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (someone2639) [reply Makefile]: good point, addressed

File: Makefile [hunk 2]
```diff
@@ -305,9 +305,11 @@ ifeq ($(filter clean distclean print-%,$(MAKECMDGOALS)),)
...
   NOEXTRACT ?= 0
   ifeq ($(NOEXTRACT),0)
-    DUMMY != $(PYTHON) extract_assets.py $(VERSION) >&2 || echo FAIL
-    ifeq ($(DUMMY),FAIL)
-      $(error Failed to extract assets from US ROM)
+    ifneq (,$(shell python3 tools/detect_baseroms.py us))
+      DUMMY != $(PYTHON) extract_assets.py us >&2 || echo FAIL
+      ifeq ($(DUMMY),FAIL)
+        $(error Failed to extract assets from US ROM)
+      endif
```
Comment (gheskett) [inline Makefile:312]: What happens now if you try to compile with no baserom? Does it still throw a useful warning or does it just attempt to compile with no assets?

Comment (someone2639) [reply Makefile]: addressed

**Pull request merged**